File: AtCoder/ARC_old/029/cpp/A_Takahashi_and_Meat.cpp
#include <bitset>
#include <iostream>
#include <stdio.h>

#define MAX_N 4

using namespace std;

int T[MAX_N]{};

int main() {
  int N;
  cin >> N;
  for (int i = 0; i < N; ++i) {
    cin >> T[i];
  }

  int res = 1'000'000'000;
  for (int i = 0; i < (1 << N); ++i) {
    bitset<4> b(i);
    int t0 = 0;
    int t1 = 0;
    for (int j = 0; j < N; ++j) {
      if (b.test(j)) {
        t0 += T[j];
      } else {
        t1 += T[j];
      }
    }
    res = min(res, max(t0, t1));
  }
  printf("%d\n", res);
}

File: AtCoder/ARC_old/029/python/A_Takahashi_and_Meat.py
N = int(input())
T = [int(input()) for _ in range(N)]

res = 1_000_000_000
for i in range(1 << N):
    b = format(i, '04b')[::-1]
    t0 = 0
    t1 = 0
    for j in range(N):
        if b[j] == '1':
            t0 += T[j]
        else:
            t1 += T[j]
    res = min(res, max(t0, t1))
print(res)
